From 4aba7405cc4028450810089d555d934cf2ef43fd Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Tue, 11 Nov 2025 22:32:52 -0500
Subject: [PATCH] Handle OSError in libaio loading

```

In [1]: from ctypes import (
   ...:     CDLL, CFUNCTYPE, POINTER, Union, Structure, memset, sizeof, byref, c_long,
   ...:     c_size_t, c_int64, c_short, c_int, c_uint, c_ulong, c_void_p, c_longlong,
   ...:     cast,
   ...: )

In [2]: CDLL('foo.so')
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Cell In[2], line 1
----> 1 CDLL('foo.so')

File ~/miniforge3/envs/dev/lib/python3.12/ctypes/__init__.py:379, in CDLL.__init__(self, name, mode, handle, use_errno, use_last_error, winmode)
    376 self._FuncPtr = _FuncPtr
    378 if handle is None:
--> 379     self._handle = _dlopen(self._name, mode)
    380 else:
    381     self._handle = handle

OSError: foo.so: cannot open shared object file: No such file or directory
```
---
 libaio/libaio.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libaio/libaio.py b/libaio/libaio.py
index 2058977..fa0898f 100644
--- a/libaio/libaio.py
+++ b/libaio/libaio.py
@@ -152,7 +152,7 @@ class io_event(Structure):
 
 try:
     libaio = CDLL('libaio.so.1t64')
-except FileNotFoundError:
+except (OSError, FileNotFoundError):
     libaio = CDLL('libaio.so.1')
 # pylint: disable=unused-argument
 def _raise_on_negative(result, func, arguments):
